File: Find All Duplicates in an Array/Find_All_Duplicates_in_an_Array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        mpp = {}
        for i in nums:
            mpp[i] = mpp.get(i, 0) + 1
        ans = []
        for key, value in mpp.items():
            if value == 2:
                ans.append(key)
        return ans
        

File: Find All Duplicates in an Array/Find_All_Duplicates_in_an_Array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        unordered_map<int,int> mpp;
        for(auto i: nums) {
            mpp[i]++;
        }
        vector<int> ans;
        for(auto i: mpp) {
            if(i.second == 2) ans.push_back(i.first);
        }
        return ans;
    }
};